client/client.go

```diff
@@ -253,30 +245,39 @@ func (c httpMetricClient) retryWithBackoff(
 	for i := 0; i < c.maxRetries; i++ {
 
 		var uploadURL, hash string
-		uploadURL, hash, err = c.GetUploadURL(metricFile, metricSampleURL, agentVersion, UID)
+		uploadURL, hash, err = c.GetUploadURL(metricFile, metricSampleURL, agentVersion, UID, i)
 		if err != nil {
 			log.Debugf("Client proxy or deployment YAML may be misconfigured.  Please check your client settings.")
 			log.Errorf("error encountered while retrieving upload location: %v", err)
 			continue
 		}
 
+		var awsRequestID, statusMessage string
+		var responseDump []byte
+		var dumpErr error
 		resp, err = c.buildAndDoRequest(metricFile, uploadURL, agentVersion, UID, hash)
-
-		if c.verbose {
-			reponseDump, err := httputil.DumpResponse(resp, true)
-			if err != nil {
-				log.Errorln(err)
+		if resp != nil {
+			awsRequestID = resp.Header.Get("X-Amz-Request-Id")
+			statusMessage = resp.Status
+			responseDump, dumpErr = httputil.DumpResponse(resp, true)
+			if dumpErr != nil {
+				log.Errorln(dumpErr)
 				continue
 			}
-			log.Infoln(string(reponseDump))
 		}
 
 		if err != nil && strings.Contains(err.Error(), "Client.Timeout exceeded") {
 			time.Sleep(getSleepDuration(i))
+			log.Infof("Put S3 Retry %d: Failed to put data to S3, Status: %s X-Amzn-Requestid: %s", i,
```

Comment:
```suggestion
			log.Infof("Put S3 Retry %d: Failed to put data to S3 due to request timeout, Status: %s X-Amzn-Requestid: %s", i,
```

Reply:
Seems like all of these should be at least warn level

Reply:
👍